linked_list/linked_list.py

```diff
@@ -11,65 +11,102 @@ class LinkedList:
+                current = current.next
+        return False
 
-    # method that returns the length of the singly linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+	# Time Complexity: O(n)
+    # Space Complexity: O(1)
     def length(self):
-        pass
+        if self.head is None:
+            return 0
+        count = 1
+        current = self.head
+        while current.next:
+            count += 1
+            current = current.next
+        return count
 
-    # method that returns the value at a given index in the linked list
-    # index count starts at 0
-    # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+	# Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_at_index(self, index):
-        pass
+        if self.head is None:
+            return None
+        index_count = 0
+        current = self.head
+        while index_count < index and current.next:
+            current = current.next
+            index_count += 1
+        if index_count == index:
+            return current.value
+        return None
 
-    # method that returns the value of the last node in the linked list
-    # returns None if the linked list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_last(self):
-        pass
+        if self.head is None:
+            return None
+        current = self.head
+        while current.next:
+            current = current.next
+        return current.value
 
-    # method that inserts a given value as a new last node in the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+	# Time Complexity: O(n)  --the cost of getting there
+    # Space Complexity: O(1)
     def add_last(self, value):
-        pass
+        if self.head is None:
+            self.add_first(value)
+            return
+        current = self.head
+        while current.next:
+            current = current.next
+        current.next = Node(value)
 
-    # method to return the max value in the linked list
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     # returns the data value and not the node
     def find_max(self):
-        pass
+        if self.head is None:
+            return None
+        current = self.head
+        max_value = current.value
+        while current:
+            if max_value < current.value:
+                max_value = current.value
+            current = current.next
+        return max_value
 
-    # method to delete the first node found with specified value
-    # Time Complexity: ?
-    # Space Complexity: ?
+	# Time Complexity: O(n)
+    # Space Complexity: O(1)
     def delete(self, value):
-        pass
+        if self.head is None:
+            return None
+        current = self.head
+        if current.value == value:
+            self.head = current.next
+        while current.next:

```

Comment:
I just want to note that this could cause the head to be deleted and then another following value which is also equal to value. 

Might be an unexpected result.